core/cap-0046-01.md

```diff
@@ -350,21 +415,19 @@ These are chosen based on two criteria:
   - Reasonably-foreseeable use in a large number of smart contracts.
   - Widely-available implementations with efficient immutable forms.
 
-In addition, _values_ are constrained by the ability to be packed into a 64-bit tagged disjoint union. Special cases for common small values such as symbols, booleans, 32-bit integers, status codes and small bitsets are provided on the basis of presumed utility in a variety of contexts. 
+In addition, _values_ are constrained by the ability to be packed into a 64-bit tagged disjoint union. Special cases for common small values such as symbols, booleans, integer types and error codes are provided on the basis of presumed utility in a variety of contexts. 
 
-The value tagging scheme is arranged into two levels -- an primary single-bit tag followed by a secondary 3-bit tag in one of the two primary cases -- in order to facilitate storing positive 64-bit integers in one of the primary cases, without overflowing to an object. We observe that the majority of 64-bit values in the current ledger are positive, representing (for example) asset amounts, time points and sequence numbers.
+Implementations of the map and vector object types are based on Rust's standard vector type, are always precisely sized to their data and immutable once constructed. The map type is a sorted vector of key-value pairs that is binary searched during map lookup, but otherwise lacks any advanced structure. 
 
-Implementations of the map and vector object types are based on design techniques from the functional language community, specifically [Relaxed-Radix-Balanced vectors (RRBs)](https://dl.acm.org/doi/10.1145/2784731.2784739) and [Hash Array Mapped Tries (HAMTs)](https://en.wikipedia.org/wiki/Hash_array_mapped_trie). Both of these data types support efficient "modifying copies" that produce new data structures from updates applied to old ones, while sharing most of the memory and substructure of the old object with the new one.
+Earlier versions of this CAP suggested the use of container objects with "shared substructure" such as HAMTs, functional red-black trees or RRBs. These were used early in Soroban's development, but it was observed that most host objects were small due to pressure from the persistent storage system and transaction system, and the overhead of objects with shared substructure exceeded the cost of a simpler approach of merely duplicating objects in full every time they are modified. As a result, the simpler approach was adopted.
 
 ### Rationale for separate XDR and host forms
```

Comment:
This section gives a good further description of the semantics difference between XDR and host form, but I don't think it really motivates why the host needs to store the objects in the host form instead XDR. 
I think one reason (other than the guest requires 64-bit integer args) is the performance inside the host. Host object handles can be cloned around cheaply, and containers of host objects are cheap and have predictable costs (map keys only have 64 bits)?

Reply:
Done